polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -658,6 +659,47 @@ Value castToIndex(Value val) {
                                             builder.getIndexType(), val);
 }
 
+/// Unroll the given loop \p loop by \p factor.
+void unrollByFactor(LoopLikeOpInterface loop, unsigned factor) {
+  if (factor == 1)
+    return;
+
+  // Lowering affine for loop to scf for loop, as not all affine loops can be
+  // unrolled. Some affine loops (e.g., min expression upper bound) cannot be
+  // unrolled, as the lower bound of the cleanup loop cannot be expressed as an
```

Comment:
cleanup -> residual

Note: IT would be possible to modify the unrolling facilities in  mlir to teach it to unroll affine loops by:
  - unroll the main affine loop
  - create an scf.for loop for the residual iterations
However that might not be a good idea in some situations because we would inject a dependency on the SCF dialect (original loop is affine but a non-affine loop is created). 

I discussed this option with @whitneywhtsang offline and we agreed that is arguably better to transform the affine loop into an scf loop here.

Reply:
done.